### ISSUE TITLE
[cssom-1] Serialization of feature values

### ISSUE THREAD

**Issue by cdoublev** [w3c/csswg-drafts#11041]
[*Serializing Media Feature Values*](https://drafts.csswg.org/cssom-1/#serializing-media-feature-values) is poorly defined.

This is not really important but hey, it could simply be replaced with [serializing a CSS value](https://drafts.csswg.org/cssom-1/#serialize-a-css-value): keywords in lowercase, base 10 numbers, etc.

In current version of Chrome and FF:

| Input                   | Chrome                  | FF                      | Expected                |
| ----------------------- | ----------------------- | ----------------------- | ----------------------- |
| `orientation: PORTRAIT` | `orientation: portrait` | `orientation: portrait` | `orientation: portrait` |
| `aspect-ratio: 1`       | `aspect-ratio: 1 / 1`   | `aspect-ratio: 1 / 1`   | `aspect-ratio: 1 / 1`   |
| `width: 0`              | `width: 0`              | `width: 0px`            | `width: 0px`            |
| `width: 1in`            | `width: 1in`            | `width: 1in`            | `width: 96px`           |
| `width: calc(1px * 2)`  | `width: calc(2px)`      | `width: calc(2px)`      | `width: calc(2px)`      |
| `color: 1.0`            | `color: 1.0`            | `color: 1.0`            | `color: 1`              |
| `color: 1e1`            | `color: 1e1`            | `color: 1e1`            | `color: 10`             |

**Comment by Loirooriol** [comment]
Lengths don't seem to serialize with the canonical unit at declaration-value time:

```css
var div = document.createElement("div");
div.style.width = "1in";
div.style.width; // "1in"
```

Also the value of the `color` feature should apparently be an `<integer>`, so `1.0` in invalid, and see #10238 for `1e1`.

**Comment by cdoublev** [comment]
Thanks, I replaced `color: 1.0` with `width: 1.0px` and removed the row with `color: 1e1`.

However, [CSSOM](https://drafts.csswg.org/cssom-1/#serialize-a-css-component-value) wants dimensions to serialize with the *"`<number>` [component] followed by the unit in its canonical form"*.

**Comment by Loirooriol** [comment]
It says

> Probably should distinguish between declared and computed / resolved values.

**Comment by cdoublev** [comment]
Right, *"in its canonical form"* does not mean *"in its canonical unit"*. I removed the row (thanks again).
